component/common/config/types.go

```diff
@@ -0,0 +1,153 @@
+// Package config contains types from github.com/prometheus/common/config, but changes them to be compatible with river
+package config
+
+import (
+	"net/url"
+
+	"github.com/prometheus/common/config"
+)
+
+// HTTPClientConfig mirrors config.HTTPClientConfig
+type HTTPClientConfig struct {
+	BasicAuth       *BasicAuth     `river:"basic_auth,block,optional"`
+	Authorization   *Authorization `river:"authorization,block,optional"`
+	OAuth2          *OAuth2Config  `river:"oauth2,block,optional"`
+	BearerToken     Secret         `river:"bearer_token,attr,optional"`
```

Comment:
This needs to use `rivertypes.Secret` instead of `string` to avoid leaking the value at /debug/config.